src/encoded/static/components/schema.js

```diff
@@ -627,7 +629,12 @@ globals.contentViews.register(SchemaPage, 'JSONSchema');
 // add a new object of that type if you're logged in.
 const AllSchemasPage = (props, reactContext) => {
     const { context } = props;
-    const loggedIn = !!(reactContext.session && reactContext.session['auth.userid']);
+    const reactContextUser = reactContext.session_properties ? reactContext.session_properties.user : null;
+    const canAddSchema = !!(
+        reactContextUser && reactContextUser.lab && reactContextUser.lab.status === 'current' &&
+        reactContextUser.submits_for && reactContextUser.submits_for.length > 0 &&
+        reactContextUser.submits_for.includes(reactContextUser.lab['@id'])
+    );
```

Comment:
Looks like you can use globals.getRoles() which does most of the same tests and check for submitter and admin here, but I won’t gate the code review on it since this does the job, but it duplicates some code.